R/DIFFUSE.R

```diff
@@ -0,0 +1,266 @@
+#' 'insulation' parameter, 3) calculate matrix E as F*D, where D is a diagonal matrix where every
+#' element (i,i) is the 'heat' score/gene score for vertex(i), 4) from each E(i,j) that is not
+#' equal to 0, assign the value E(i,j) to the directional edge going from vertex j to vertex i.
+#' A more precise definition of the model E=Beta*inverse(I-(1-Beta)*W)*D is a random walk where,
+#' if one starts from node j, the probability of moving onto another given node will be
+#' (1-Beta)*1/deg(j), where Beta is the probability of restarting from the start node. If this
+#' process is repeated ad infinitum, an equilibrium distribution of F=Beta*inverse(I-(1-Beta)*W)
+#' is reached, where in each column j of matrix F, a vector of probabilities is given for landing
+#' on each element (i) in the column; i.e. element i of column j represents the probability of
+#' reaching node i if a random walk starts from node j. Thus, element E(i,j) represents the
+#' probability of reaching node i from node j when at equilibrium multiplied by the initial
+#' 'heat' (gene score) put on node j. For all E(i,j) where there exists a directed edge from
+#' j to i, edge j -> i is equal to the value of E(i,j).
+#'
+#' @export
+
+DIFFUSE <- function(AGG = NULL, algorithm = "Leis", param = list(getOption("rete.beta")),
+                    silent = FALSE, noLog = FALSE) {
+        startTime <- Sys.time()
+
+        consoleVect <- paste("Diffuse started at", Sys.time(), sep = "")
+        if (!silent) {
+            print(consoleVect) #print beginning of function to console
+        }
+#========= Checking Arguments =============================
+        consoleVect <- "Checking Arguments"
+        if (!silent) {
+            print(consoleVect) #print beginning of function to console
+        }
+
+        #Check if AGG is an igraph object, and check if it's directed
+        if (!(igraph::is.igraph(AGG) && igraph::is.directed(AGG))) {
+            stop("Error: AGG must be an iGraph object with directed edges")
+        }
+
+
+        #Check other argmuments
+        check1 <- .checkArgs(algorithm,character(length = 1),checkSize = TRUE)
+        check2 <- .checkArgs(silent, logical(length = 1), checkSize = TRUE)
+        check3 <- .checkArgs(noLog, logical(length = 1), checkSize = TRUE)
+        check4 <- .checkArgs(param, list())
+
+        if (length(c(check1,check2,check3,check4)) > 0 ) {
+            stop("Error: incorrect argument(s) for algorithm, silent, noLog, and/or param.
+                 See documentation for DIFFUSE function")
+        }
+
+        #Extract AGG metadata for writing to log
+        metaAGG <- attr(AGG,"meta")
+           #Assign AGG vertices and edges to data frames
+           AGGverts<- igraph::as_data_frame(AGG, "vertices")
+           AGGedges<- igraph::as_data_frame(AGG, "edges")
+
+           sizeEdges <- nrow(AGGedges) #Number of edges, used for indexing purposes
```

Comment:
These names are not well chosen since "size" and "order" have a special meaning for a graph. I would change them to "nVerts" and "nEdges" - that's the common way and thus more readable.